packages/plugin-npm/sources/npmHttpUtils.ts

```diff
@@ -242,7 +397,7 @@ function shouldAuthenticate(authConfiguration: MapLike, authType: AuthType) {
   }
 }
 
-async function whoami(registry: string, headers: {[key: string]: string} | undefined, {configuration}: {configuration: Configuration}) {
+async function whoami(registry: string, headers: {[key: string]: string | undefined} | undefined, {configuration}: {configuration: Configuration}) {
```

Comment:
```suggestion
async function whoami(registry: string, headers: {[key: string]?: string} | undefined, {configuration}: {configuration: Configuration}) {
```

Reply:
Seems like dynamic keys can't be optional, it's a syntax error.

Reply:
Weird, I was sure to have used something similar somewhere in the past ... probably thinking of something else, nevermind